### PR TITLE
Update statsd-client types to version 0.4.0

### PR DESCRIPTION
**Pull request by chris-codaio**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/msiebuhr/node-statsd-client
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/statsd-client/index.d.ts [hunk 1]
```diff
@@ -79,24 +88,53 @@ interface HttpOptions extends CommonOptions {
...
+    timeByUrl?: boolean;
+}
+
 declare class StatsdClient {
     constructor(options: TcpOptions | UdpOptions | HttpOptions);
 
-    counter(metric: string, delta: number): void;
-    increment(metric: string, delta?: number): void;
-    decrement(metric: string, delta?: number): void;
+    counter(metric: string, delta: number, tags?: {}): this;
```
Comment (minestarks) [inline types/statsd-client/index.d.ts]: Any type is assignable to `{}`, including primitives like `number`. So this is the same as saying `tags?: any`. To be clearer, I suggest changing the type to `any` (or `object`, if your intention is to disallow primitive types).

File: types/statsd-client/index.d.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
-// Type definitions for statsd-client v0.1.0
+// Type definitions for statsd-client v0.4.0
 // Project: https://github.com/msiebuhr/node-statsd-client
 // Definitions by: Peter Kooijmans <https://github.com/peterkooijmans>
+//                 Christopher Eck <https://github.com/chrisleck>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
 
-
+import * as express from "express";
```
Comment (minestarks) [inline types/statsd-client/index.d.ts]: Just a suggestion, but if you're only using `RequestHandler` you can just import that. I think it's more readable that way.

**Review by minestarks — CHANGES_REQUESTED** [review]

File: types/statsd-client/index.d.ts (hunk 2 above)
Comment (chrisleck) [reply types/statsd-client/index.d.ts]: I'm using `Request`, `Response`, and `RequestHandler`.

File: types/statsd-client/index.d.ts (hunk 1 above)
Comment (chrisleck) [reply types/statsd-client/index.d.ts]: Updated to a real hash type.

**Comment by chrisleck** [comment]
Thanks for the review.  Updates applied.  PTAL, @minestarks.

**Review by minestarks — APPROVED** [review]

**Pull request merged**